### ISSUE TITLE
Can't import chart props type: Module '"recharts"' has no exported member 'LineChartProps'

### ISSUE THREAD

**Issue by tran-simon** [recharts/recharts#3619]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/recharts-issue-template-forked-qfp8cm?file=/src/index.tsx)

### Steps to reproduce
import "LineChartProps" or any other props type

### What is expected?
You should be able to import a props type in order to create custom components and extend the props type

### What is actually happening?
 Module '"recharts"' has no exported member 'LineChartProps'.

| Environment | Info |
|---|---|
| Recharts | v2.5.0 |
| React | 18.2.0 |
| System | codesandbox |
| Browser | Chrome |

**Comment by ckifer** [comment]
Hi, the codesandbox you provided is working unless I'm missing something

**Comment by tran-simon** [comment]
Is there a reason why it's not simply exported like other types (XAxisProps, BarProps...)? I know that ApexChart lets you import pretty much anything

**Comment by ckifer** [comment]
Caveat that I didn't write any of this code and those who did are long gone, so I generally don't know. 

My fear however is that this file is relatively unwieldy and sanctioning taking a dependency on a type within it is asking for changes that aren't intending to be breaking to break consumers accidentally down the line.

**Comment by ckifer** [comment]
CC @nikolasrieble in case you have a different perspective on this

**Comment by nikolasrieble** [comment]
The `CategoricalChart` is an implementation detail and not a component that is exported / usable. That is why I would not suggest to expose CategoricalChartProps. The LineChart on the other hand is, and its prop type should be exported.

**Comment by ckifer** [comment]
It doesn't have a prop type though, it's prop type is CategorialChartProps

**Comment by nikolasrieble** [comment]
> It doesn't have a prop type though, it's prop type is CategorialChartProps (along with the rest of the categorical charts)
Right, so maybe we just alias it for now, and expose it as LineChartProps?

**Comment by tran-simon** [comment]
Also, while `BarProps` is exported, I can't seem to use it like I would expect?

https://codesandbox.io/s/recharts-issue-template-forked-652ynn?file=/src/index.tsx

```
const MyBar = (props: BarProps) => {
  return <Bar {...props} />;
};
```

I get this error on `<Bar `:

> No overload matches this call.
  Overload 1 of 2, '(props: Props | Readonly<Props>): Bar', gave the following error.
    Type '{ string?: string | number; children?: ReactNode; dangerouslySetInnerHTML?: { __html: string; }; onCopy?: AdaptChildClipboardEventHandler<any, SVGPathElement>; ... 493 more ...; left?: number; }' is not assignable to type 'IntrinsicClassAttributes<Bar>'.
      Types of property 'ref' are incompatible.
        Type 'LegacyRef<SVGPathElement>' is not assignable to type 'LegacyRef<Bar>'.
          Type '(instance: SVGPathElement) => void' is not assignable to type 'LegacyRef<Bar>'.
            Type '(instance: SVGPathElement) => void' is not assignable to type '(instance: Bar) => void'.
              Types of parameters 'instance' and 'instance' are incompatible.
                Type 'Bar' is missing the following properties from type 'SVGPathElement': addEventListener, removeEventListener, pathLength, getPointAtLength, and 272 more.
  Overload 2 of 2, '(props: Props, context: any): Bar', gave the following error.
    Type '{ string?: string | number; children?: ReactNode; dangerouslySetInnerHTML?: { __html: string; }; onCopy?: AdaptChildClipboardEventHandler<any, SVGPathElement>; ... 493 more ...; left?: number; }' is not assignable to type 'IntrinsicClassAttributes<Bar>'.
      Types of property 'ref' are incompatible.
        Type 'LegacyRef<SVGPathElement>' is not assignable to type 'LegacyRef<Bar>'.

**Comment by ckifer** [comment]
Yeah unfortunately there's a lot of issues with types due to stuff like this https://github.com/recharts/recharts/blob/36b75d49d8adc4fc7459faf28e45cc579d2f1a20/src/cartesian/Bar.tsx#LL97C98-L97C106

**Comment by mortoni** [comment]
I think my issue has to do with this, and I don't want to create a reproduction link, however, I am using Recharts 2.6.2 and I am getting `Namespace 'React' has no exported member 'JSX'.ts(2694)`. They are trying to use React.JSX.Element, when I think it should be JSX.Element.

I can't get vite to build my application anymore, I have to downgrade Recharts to 2.5.0.

**Comment by ckifer** [comment]
@mortoni not sure what you mean - this issue doesn't have anything to do with JSX.Element nor are there references to React.JSX.Element in the code. Can you give any more details on your issue?

**Comment by nikolasrieble** [comment]
> I think my issue has to do with this, and I don't want to create a reproduction link, however, I am using Recharts 2.6.2 and I am getting `Namespace 'React' has no exported member 'JSX'.ts(2694)`. They are trying to use React.JSX.Element, when I think it should be JSX.Element.
> 
> I can't get vite to build my application anymore, I have to downgrade Recharts to 2.5.0.

@mortoni This does sound like a separate issue. Please create a new issue instead of commenting on this one.
